### ISSUE TITLE
FactorInt for big.Int ?

### ISSUE THREAD

**Issue by equal-l2** [cznic/mathutil#12]
It would be great if `FactorInt` supports `big.Int`.

**Comment by cznic** [comment]
Which factoring algorithm it should implement? How should the API look like?

**Comment by equal-l2** [comment]
I have no concrete idea, but we can't use the same algorithm as `FactorInt(uint32)`(trial division with precalculated primes) for `big.Int`, right?

**Comment by cznic** [comment]
> I have no concrete idea, but we can't use the same algorithm as FactorInt(uint32)(trial division with precalculated primes) for big.Int, right?

Yes, that's definitely possible. What range of integers on input do you want to be able to factor using this approach?
